libfreerdp/core/nego.c

```diff
@@ -332,7 +333,7 @@ BOOL nego_transport_connect(rdpNego* nego)
 BOOL nego_transport_disconnect(rdpNego* nego)
 {
 	if (nego->TcpConnected)
-		transport_disconnect(nego->transport);
+		nego->transport->context->update->io->TransportDisconnect(nego->transport);
```

Comment:
I don´t like the exposure of the internal structure of `rdpTransport` you´re doing here.
We are working on moving all module/components to be self contained with only an opaque structure exposed, this moves in the opposite direction.